django_project/minisass_authentication/models/yoma.py

```diff
@@ -0,0 +1,261 @@
+import requests
+import uuid
+from constance import config
+from django.db import models
+from django.contrib.auth import get_user_model
+from django.utils import timezone
+from datetime import timedelta
+
+User = get_user_model()
+
+
+class YomaCountry(models.Model):
+    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
+    name = models.CharField(max_length=255)
+    code_alpha2 = models.CharField(max_length=2, unique=True)
+    code_alpha3 = models.CharField(max_length=3, unique=True)
+    code_numeric = models.CharField(max_length=3, unique=True)
+
+
+class YomaToken(models.Model):
+    """
+    Store YOMA OAuth2 token data for authenticated users.
+
+    This model stores the access token, refresh token, and related metadata
+    received from YOMA's OAuth2 token endpoint.
+    """
+    user = models.OneToOneField(
+        User,
+        on_delete=models.CASCADE,
+        related_name='yoma_token',
+        help_text="User associated with this YOMA token"
+    )
+
+    access_token = models.TextField(
+        help_text="YOMA access token for API requests"
+    )
+
+    refresh_token = models.TextField(
+        blank=True,
+        null=True,
+        help_text="YOMA refresh token for obtaining new access tokens"
+    )
+
+    id_token = models.TextField(
```

Comment:
@zamuzakki can you investigate and check whether it is safe for us to store the id token as is. Or do we need to encrypt it?

Reply:
@dimasciput Yeah right, it's better be encrypted. I shall update it.